src/Raven.Server/ServerWide/ClusterStateMachine.License.cs

```diff
@@ -391,72 +393,149 @@ void AssertAutoIndexesCount()
-            }
+            var putSubscriptionCommand = (T)JsonDeserializationCluster.Commands[typeof(T).Name](command);
+            putSubscriptionCommandsList.Add(putSubscriptionCommand);
+
+            if (includeRevisions == false &&
+                putSubscriptionCommand.Query.Contains(DocumentSubscriptions.IncludeRevisionsRQL))
+                includeRevisions = true;
         }
 
-        if (serverStore.LicenseManager.LicenseStatus.HasRevisionsInSubscriptions == false &&
-            command.Query.Contains(DocumentSubscriptions.IncludeRevisionsRQL))
+        if (AssertSubscriptionRevisionFeatureLimits(serverStore, includeRevisions, context) == false)
+            return putSubscriptionCommandsList;
+
+        var subscriptionsNamesPerDatabase = new Dictionary<string, List<string>>();
+        foreach (var command in putSubscriptionCommandsList)
         {
-            if (CanAssertLicenseLimits(context, minBuildVersion: MinBuildVersion60000) == false)
-                return;
+            if (subscriptionsNamesPerDatabase.TryGetValue(command.DatabaseName, out _) == false)
+                subscriptionsNamesPerDatabase.Add(command.DatabaseName, new List<string>());
 
-            throw new LicenseLimitException(LimitType.Subscriptions, "Your license doesn't include the subscription revisions feature.");
+            subscriptionsNamesPerDatabase[command.DatabaseName].Add(command.SubscriptionName);
         }
 
-        long GetSubscriptionsCount()
+        if (AssertNumberOfSubscriptionsPerClusterLimits(serverStore, items, context, subscriptionsNamesPerDatabase))
+            return putSubscriptionCommandsList;
+
+        AssertNumberOfSubscriptionsPerDatabaseLimits(serverStore, items, context, subscriptionsNamesPerDatabase);
+        return putSubscriptionCommandsList;
+    }
+
+    private bool AssertNumberOfSubscriptionsPerDatabaseLimits(
+        ServerStore serverStore,
+        Table items,
+        ClusterOperationContext context,
+        IReadOnlyDictionary<string, List<string>> subscriptionsNamesPerDatabase)
+    {
+        foreach ((string databaseName, List<string> subscriptionsNames) in subscriptionsNamesPerDatabase)
         {
-            long count = 0;
+            var subscriptionsCount = GetSubscriptionsCountForDatabase(context.Allocator, items, databaseName, subscriptionsNames);
 
-            foreach (var name in GetDatabaseNames(context))
-            {
-                count += GetSubscriptionsCountForDatabase(context.Allocator, items, name);
-            }
+            var maxSubscriptionsPerDatabase = serverStore.LicenseManager.LicenseStatus.MaxNumberOfSubscriptionsPerDatabase;
+            if (maxSubscriptionsPerDatabase is not >= 0)
+                return false;
 
-            return count;
+            if (subscriptionsCount + subscriptionsNames.Count > maxSubscriptionsPerDatabase == false)
+                return false;
```

Comment:
shouldn't we use `continue` here?

Reply:
I structured the logic of the assertion differently...